src/wavm.cpp

```diff
@@ -193,9 +227,49 @@ ExecutionResult WavmEngine::execute(
   evmc_message const& msg,
   bool meterInterfaceGas
 ) {
+
+  // clear stuff from previous run
+  wavm_host_module::exceptionIdx = wavm_host_module::HeraExceptionsEnum::None;
+  wavm_host_module::exceptionString = "";
+  // hope and prayer
+  Runtime::collectGarbage();
+
+  // execute the contract
   ExecutionResult result = internalExecute(context, code, state_code, msg, meterInterfaceGas);
-  // And clean up mess left by this run.
+
+  // clean up this run, this is done here after leaving the scope of internalExecute()
   Runtime::collectGarbage();
+
+  // re-throw exception if there was one
+  if (wavm_host_module::exceptionIdx != wavm_host_module::HeraExceptionsEnum::None){
```

Comment:
maybe simplify using a helper?

Reply:
Could do it if you insist. But `hera_execute()` does a similar long `switch-case` inside of an `if` without a helper function.